pkg/glusterfs/Dockerfile

```diff
@@ -1,22 +1,71 @@
 # Copyright 2018 The Gluster CSI Authors.
 
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-#You may obtain a copy of the License at
-
-#    http://www.apache.org/licenses/LICENSE-2.0
+# Licensed under GNU LESSER GENERAL PUBLIC LICENSE Version 3, 29 June 2007
+# You may obtain a copy of the License at
+#    https://opensource.org/licenses/lgpl-3.0.html
 
 # Unless required by applicable law or agreed to in writing, software
 # distributed under the License is distributed on an "AS IS" BASIS,
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM centos:7.5.1804
 
-# Copy glusterfs-csi-driver from build directory
-COPY ./build/glusterfs-csi-driver /glusterfs-csi-driver
+# Establish the base image that will be used
+ARG BASE_IMAGE="centos:7.5.1804"
+
+#-- Build phase
+FROM ${BASE_IMAGE} AS build
+
+ENV GOPATH="/go/" \
+    SRCDIR="/go/src/github.com/gluster/gluster-csi-driver/"
+
+RUN yum install -y \
+    git \
+    golang
+
+# Install dep
+RUN mkdir -p /go/bin
+RUN curl -L https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+
+# Vendor dependencies
+COPY Gopkg.lock Gopkg.toml "${SRCDIR}"
+WORKDIR "${SRCDIR}"
+RUN /go/bin/dep ensure -v -vendor-only
+
+# Build executable
+COPY . "${SRCDIR}"
+RUN CGO_ENABLED=0 GOOS=linux go build -ldflags '-extldflags "-static"' -o /glusterfs-csi-driver cmd/glusterfs/main.go
+
+# Ensure the binary is statically linked
+RUN ldd /glusterfs-csi-driver | grep -q "not a dynamic executable"
+
+
+#-- Final container
+FROM ${BASE_IMAGE} AS final
+
+# Install dependencies
+RUN yum -y install centos-release-gluster && \
```

Comment:
why we require `centos-release-gluster` package here?

Reply:
It enables the CentOS storage sig repos so we can get current gluster packages instead of the default (old) ones in the main repo.

Reply:
@Madhu-1 @JohnStrunk cant we directly install latest stable version from download.gluster.org instead of using centos version ?

Reply:
The CentOS Storage SIG provides the latest stable version. It is what we recommend to use on CentOS systems, see https://gluster.readthedocs.io/en/latest/Install-Guide/Community_Packages/